### PR TITLE
Add autofix tests

### PR DESCRIPTION
**Pull request by colinkiama**

### CODE REVIEW

**Comment by colinkiama** [comment]
Not all implemented fixes have been added to the tests. I found some issues with them during testing. 

These can be fixed later though. The main point of this PR is setting up a structure for testing fixes.

**Comment by colinkiama** [comment]
@dar5hak Oh hey 👋 , I’m pretty much blocked until this PR goes through. 

Please merge whenever your ready.

**Review by dar5hak — APPROVED** [review]

**Comment by dar5hak** [comment]
Sorry, man! I missed the notification for this PR. Thanks for tagging.

Two thoughts:
- Should we be writing the recursive file copying logic ourselves? Doesn't GIO have something built-in for this?
- Can you run lint on the test file itself?

**Comment by colinkiama** [comment]
Regarding recursive File copying, strangely no 🤔: https://valadoc.org/gio-2.0/GLib.File.copy_async.html

The usage of the File enumerator seems to be the widely accepted way: https://stackoverflow.com/questions/16453739/how-do-i-recursively-copy-a-directory-using-vala

I could run `vala-lint` on the test files themselves. This would involve me removing the first two lines of every copied test file though.

Also, is it a good idea to allow these tests potentially fail from  a bug that's from a completely different part of the the program? Wouldn't it be in our best interest to keep the auto-fix tests within the scope of the auto-fix feature?

**Comment by dar5hak** [comment]
Oh no, I just meant to lint `AutoFixTest.vala`, not the files you are using to lint. Typed that before breakfast, sorry. 😅

**Comment by dar5hak** [comment]
We are good to merge after that, and I'll relay the PR to upstream.

File: test/AutoFixTest.vala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    public struct AutoFixFileData {
+        File file;
+        string name;
+        Vala.ArrayList<ValaLint.FormatMistake?> mistakes;
+    }
+
+    public bool copy_recursive (GLib.File src, GLib.File dest,
+        GLib.FileCopyFlags flags = GLib.FileCopyFlags.NONE, GLib.Cancellable? cancellable = null) throws GLib.Error {
+      GLib.FileType src_type = src.query_file_type (GLib.FileQueryInfoFlags.NONE, cancellable);
+      if ( src_type == GLib.FileType.DIRECTORY ) {
```
[the remark this answers is not in the document]
Comment (colinkiama) [reply test/AutoFixTest.vala:36]: Stackoverflow :sweat_smile:

**Review by dar5hak — APPROVED** [review]

**Pull request merged**